omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/QuickVisitFragmentController.java

```diff
@@ -38,16 +50,33 @@
 	public FragmentActionResult create(@SpringBean("adtService") AdtService adtService,
 	                                   @RequestParam("patientId") Patient patient,
 	                                   @RequestParam("locationId") Location location, UiUtils uiUtils,
+									   @RequestParam(value = "selectedTypeId", required = false) VisitType selectedType,
 	                                   UiSessionContext emrContext,
 	                                   HttpServletRequest request) {
 
 		VisitDomainWrapper activeVisit = adtService.getActiveVisit(patient, location);
+		BindingResult bindingResult = null;
 		if (activeVisit != null) {
 			return new FailureResult(uiUtils.message("coreapps.activeVisits.alreadyExists"));
 		}
 
         Visit visit = adtService.ensureVisit(patient, new Date(), location);
 
+		if(selectedType != null){
+			//set visit type
+			visit.setVisitType(selectedType);
+
+			// manually handle the attribute parameters
+			List<VisitAttributeType> attributeTypes = Context.getVisitService().getAllVisitAttributeTypes();
+			WebAttributeUtil.handleSubmittedAttributesForType(visit, bindingResult, VisitAttribute.class, request, attributeTypes);
+			try {
+				Context.getVisitService().saveVisit(visit);
+			}
+			catch (APIException e) {
+				request.getSession().setAttribute(WebConstants.OPENMRS_ERROR_ATTR, "Visit.save.error");
```

Comment:
Nope. The message won't be overwritten instead both error and success messages will be returned. Nice catch!